packages/core/src/routes/session/utils.ts

```diff
@@ -189,8 +189,54 @@ export const checkRequiredProfile = async (
     throw new RequestError({ code: 'user.require_email_or_sms', status: 422 });
   }
 };
+
+export const checkRequiredSignUpIdentifiers = async (identifiers: {
+  username?: string;
+  primaryEmail?: string;
+  primaryPhone?: string;
+}) => {
+  const { username, primaryEmail, primaryPhone } = identifiers;
+
+  const { signUp } = await findDefaultSignInExperience();
+
+  if (signUp.identifier === SignUpIdentifier.Username && !username) {
+    throw new RequestError({ code: 'user.require_username', status: 422 });
+  }
+
+  if (signUp.identifier === SignUpIdentifier.Email && !primaryEmail) {
+    throw new RequestError({ code: 'user.require_email', status: 422 });
+  }
+
+  if (signUp.identifier === SignUpIdentifier.Sms && !primaryPhone) {
+    throw new RequestError({ code: 'user.require_sms', status: 422 });
+  }
+
+  if (signUp.identifier === SignUpIdentifier.EmailOrSms && !primaryEmail && !primaryPhone) {
```

Comment:
```suggestion
  if (signUp.identifier === SignUpIdentifier.EmailOrSms && (!primaryEmail || !primaryPhone)) {
```
Do we need to provide BOTH email and phone when signUpIdentifier is EmailOrSms?

Reply:
Nope